sdk-design-doc-template.md

```diff
@@ -0,0 +1,83 @@
+
+## New APIs
+
+Define the names of the new APIs being added, as well as how they are interacted with.
+
+### New API #1
+
+- `Type property` : description of the property.
+    - Make note of any additional features to know about the property, such as min/max values, required/optional, etc.
+    - Provide generic examples of setting/getting the property in code.
+        - `NewApi setProperty(Type prop)`
+        - `Type getProperty()`
+- `NewApi function(Type1 var1, Type2 var2,...)` : description of what the function does.
+    - Make note of any additional features/ramifications of using this function (anything that may not be immediately clear to the user).
+    - `Type1 var1` : description of input variable.
+        - Make note of any additional features to know about this input, such as min/max values, etc.
+    - `Type2 var2` : description of input variable.
+
+### New API #2
+
+…
+
+---
+
+## Updated APIs
+
+APIs that already exist in the SDKs that are getting updated in some way (generally fields are being added). A brief explanation of why the API is being updated as well should be provided.
+
+### Updated API #1
+
+- `Type newProperty` : description of the new property.
+    - Make note of any additional features to know about the new property, such as min/max values, required/optional, etc.
+    - Provide generic examples of setting/getting the new property in code.
+        - `NewApi setProperty(Type prop)`
+        - `Type getProperty()`
+- `NewApi function(Type1 var1, Type2 var2,...)` : description of what the function does.
+    - Make note of any additional features/ramifications of using this function (anything that may not be immediately clear to the user).
+    - `Type1 var1` : description of input variable.
+        - Make note of any additional features to know about this input, such as min/max values, required/optional, etc.
+    - `Type2 var2` : description of input variable.
+
+### Updated API #2
+
+…
+
+---
+
+## Internal Changes
+
+If applicable, description of any internal changes to the SDKs that may impact usage that don't involve an API change. It should describe why the change is happening and its relevancy to the feature. It should describe any new expected behavior and how it differs from old behavior, as well as how users should remedy the behavior chagne.
+
+### Response Codes
+
+If applicable, include any new consensus node response codes associated with this feature. Response codes are messages that the consensus node returns when a transaction succeeds or fails.
```

Comment:
How are response codes listed? A table? What information is needed for new response codes?

Reply:
Just a list. I'll add an example.